crates/forge_app/src/orch.rs

```diff
@@ -643,13 +642,16 @@ impl<S: AgentService> Orchestrator<S> {
             turn_has_tool_calls = turn_has_tool_calls || has_tool_calls;
         }
 
+        // Update metrics in conversation
+        tool_context.with_metrics(|metrics| {
+            self.conversation.metrics = metrics.clone();
+        })?;
+
         if has_attempted_completion {
-            self.send(ChatResponse::TaskComplete { metrics: metrics.clone() })
+            self.send(ChatResponse::TaskComplete { metrics: self.conversation.metrics.clone() })
```

Comment:
If the conversation already has a summary in sync, do we need to dispatch it as a separate message? Can't we just read the summary from the conversation wherever required?